### ISSUE TITLE
TypeScript build error - no changes

### ISSUE THREAD

**Issue by cryptoB0T** [pusher/pusher-http-node#128]
I haven't made any changes anywhere in the code. I'm starting a fresh project from:
https://github.com/microsoft/TypeScript-Node-Starter

I see this in the readme but not entirely sure what it means:
> All external APIs should have types in index.d.ts.

so i  defined my own /types/pusher/index.d.ts and declared namespaces below. 

but i can't figure out how to resolve `error TS2456: Type alias 'Response' circularly references itself.`

has anybody had this issue? 

## typescript build error 

```
node_modules/pusher/index.d.ts:50:18 - error TS2694: Namespace 'http' has no exported member 'Agent'.

50     agent?: http.Agent
                    ~~~~~

node_modules/pusher/index.d.ts:138:40 - error TS2503: Cannot find namespace 'request'.

138     constructor(token: Token, request: request.Request)
                                           ~~~~~~~

node_modules/pusher/index.d.ts:148:15 - error TS2456: Type alias 'Response' circularly references itself.

148   export type Response = Response
                  ~~~~~~~~


Found 3 errors.
```

**Comment by callum-oakley** [comment]
Hi @cryptoB0T thanks for reporting this. The SDK doesn't use typescript internally, and the types have drifted out of sync with the implementation. I'll push a fix for this asap, watch this space!

In the meantime you could remove the type file from your `node_modules` to suppress the error until the types are fixed.

```
rm node_modules/pusher/index.d.ts
```

**Comment by cryptoB0T** [comment]
@callum-oakley thanks for your prompt response! 

i can get it working locally to suppress the error, but i haven't figured out how to get this deployed because heroku is running a build each time. 

i guess i could point to a local copy of the pusher module in the file or package json. would you say that's the easiest approach?

**Comment by cryptocharlie404** [comment]
@callum-oakley do you have an ETA when you expect to push a fix? 

i tried to get around this by just including it in my project, but it was just error after error. do you need more details about the type errors? thanks!

**Comment by cryptoB0T** [comment]
@callum-oakley do you have an ETA when you expect to push a fix? 

i tried to get around this by just including it in my project, but it was just error after error. do you need more details about the type errors? thanks!

**Comment by cryptoB0T** [comment]
i needed to get this working, so i included the package locally. explicitly ignore any lint errors, installed all the required dependencies. had to make a couple changes, version was including package which was not a module. and the pusher object was maybe implicitly casting a non-existent property from the comments, so i added cluster. it's working! looking forward to hearing when this is updated so i can pull in the latest. Thanks!

**Comment by callum-oakley** [comment]
Glad to hear you found a workaround. I believe #129 should be a proper fix, which I'll merge and push out just as soon as someone has taken a look at it and CI's passed.

If you're feeling so inclined, you can make sure the branch works for you by pointing to the branch in your `package.json` e.g.

```
"pusher": "pusher/pusher-http-node#repair-types"
```

**Comment by callum-oakley** [comment]
Fixed in v4.0.1. Let me know if you still see any type errors after upgrading.
